src/TypeInference/Unification.mli

```diff
@@ -41,6 +41,14 @@ type label =
   | L_Label of T.effect * T.typ * T.effrow
     (** Label type *)
 
+type unification_result =
```

Comment:
This name of type is very long, when used in qualified form: `Unification.unification_result`. Since we never open `Unification` module, I suggest to shorten it to just `result`.